consensus/src/consensus.rs

```diff
@@ -396,15 +396,17 @@ impl<R: ConsensusRpc> ConsensusClient<R> {
 
     // implements checks from validate_light_client_update and process_light_client_update in the
     // specification
-    fn verify_generic_update(&self, update: &GenericUpdate) -> Result<()> {
+    fn verify_generic_update(&self, update: &mut GenericUpdate) -> Result<()> {
         let bits = get_bits(&update.sync_aggregate.sync_committee_bits);
         if bits == 0 {
             return Err(ConsensusError::InsufficientParticipation.into());
         }
 
-        let update_finalized_slot = update.finalized_header.clone().unwrap_or_default().slot;
-        let valid_time = self.expected_current_slot() >= update.signature_slot
-            && update.signature_slot > update.attested_header.slot
+        let update_finalized_slot = match &update.finalized_header {
+            Some(header) => header.slot,
+            None => 0,
+        };
```

Comment:
```suggestion
        let update_finalized_slot = update.finalized_header.map(|header| header.slot).unwrap_or(0);
```

p.s. I didn't do format for this piece of code.

Reply:
great suggestion here